cpp/models/abm/world.cpp

```diff
@@ -166,6 +167,16 @@ int World::get_subpopulation_combined(InfectionState s, LocationType type) const
     });
 }
 
+void World::save_current_subpopulations(TimePoint& t, TimeSpan& dt)
+{
+    t += dt;
```

Comment:
You are changing the variable t here which was not done before in World::evolve. Are you sure you want to do this?